### PR TITLE
Unit Test For Caption

### PR DESCRIPTION
**Pull request by EramM805**
We will be using a dependency called check-prop-types for PropType testing. I have completed unit tests for caption. Please let me know if you have any questions/concerns/suggestions. Also please let me know if I have forgotten an edge case.

### CODE REVIEW

File: __tests__/UserCaption-test.js [hunk 1]
```diff
@@ -1,12 +1,186 @@
 import 'react-native';
 import React from 'react';
 import UserCaption from '@screens/app/Components/UserCaption/UserCaption';
+import checkPropTypes from 'check-prop-types';
+var assert = require('assert');
```
Comment (bujars) [inline __tests__/UserCaption-test.js:5]: Where is this being pulled from?

File: __tests__/UserCaption-test.js [hunk 2]
```diff
@@ -1,12 +1,186 @@
...
+describe('testing UserCaption values, should not throw errors', () => {
+  const stringCaptionTest = 'Hi World, Missed Me ?';
+  const stringUsernameTest = 'e805';
+  it('testing UserCaption Creation String', () => {
+    const wrapper = shallow(
+      <UserCaption username={stringUsernameTest} caption={stringCaptionTest} />,
+    );
+    const instance = wrapper.instance();
+    const currentUsername = instance.state.username;
+    const currentCaption = instance.state.caption;
```
Comment (bujars) [inline __tests__/UserCaption-test.js:26]: What is the reason behind giving caption a state and not just a prop? Is there anywhere where we are manipulating data? 

-- If we change to prop --> instance.props.username would be the fix

File: __tests__/UserCaption-test.js [hunk 3]
```diff
@@ -1,12 +1,186 @@
...
+      {caption: 234.5, username: 'emanasia'},
+      'prop',
+      UserCaption.username,
+    );
+    assert(
+      result ===
+        'Failed prop type: Invalid prop `caption` of type `number` supplied to `<<anonymous>>`, expected `string`.',
+    );
+  });
+});
```
Comment (bujars) [inline __tests__/UserCaption-test.js:185]: Can we combine all of these into a function, ie testPropTypes (username, caption, errormessage etc)
Since those are the default values that change with each test case?

Comment (bujars) [reply __tests__/UserCaption-test.js:185]: We can do this in a refactor process if needed ^

File: src/screens/Feed.js [hunk 4]
```diff
@@ -30,7 +30,7 @@ export default class FeedScreen extends React.Component {
     const isLiked = likers[uid] === true ? true : false; // if i liked it
     const isBookmarked = bookmarkers[uid] === true ? true : false;
     return (
-      <TouchableWithoutFeedback onPress={() => { }}>
+      <TouchableWithoutFeedback onPress={() => {}}>
```
Comment (bujars) [inline src/screens/Feed.js:33]: Property onPress is not required, and thus we can remove it (If unless it is needed for something else down the road -- Which then I recommend writing comments. )
https://reactnative.dev/docs/touchablewithoutfeedback

File: src/screens/app/Components/SettingsButtonVertical/SettingsButtonVertical.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    iconSize: 30,
+    settingsButtonVerticalIcon: 'dots-vertical',
+  };
+  handleSettingsClick = () => {};
+  render() {
+    return (
+      <IconButton
+        icon={this.state.settingsButtonVerticalIcon}
+        size={this.state.iconSize}
+        onPress={this.handleSettingsClick}
```
Comment (bujars) [inline src/screens/app/Components/SettingsButtonVertical/SettingsButtonVertical.js:15]: Why are we naming this SettingsButtonHorizontal but utilizing the settingsButtonVerticalIcon? Do we want consistency?

File: src/screens/app/Components/UserCaption/UserCaption.js [hunk 6]
```diff
@@ -1,23 +1,32 @@
...
   constructor(props) {
     super(props);
-  }
 
-  handleCommentClick = () => {};
+    this.state = {
+      caption: this.props.caption,
+      username: this.props.username,
+    };
+  }
```
Comment (bujars) [inline src/screens/app/Components/UserCaption/UserCaption.js:14]: As noted in the test file, what is the reason behind storing state? When will we need it for data manipulation?

File: src/screens/app/Components/UserCaption/UserCaption.js [hunk 7]
```diff
@@ -1,23 +1,32 @@
...
-        <Text style={styles.boldStyle}>{this.props.username}</Text>{' '}
-        <Text style={styles.captionStyles}>{this.props.caption}</Text>
+        <Text style={styles.boldStyle}>{this.state.username}</Text>{' '}
+        <Text style={styles.captionStyles}>{this.state.caption}</Text>
       </Subheading>
     );
   }
 }
+UserCaption.propTypes = {
+  caption: PropTypes.string.isRequired,
```
Comment (bujars) [inline src/screens/app/Components/UserCaption/UserCaption.js:26]: By isRequired, are we only focusing on it being a string? or a non empty string? Because if no caption is provided, of course we just leave it as the empty string ''

File: package.json [hunk 8]
```diff
@@ -35,13 +35,15 @@
     "react-native-tab-view": "^2.13.0",
     "react-native-vector-icons": "^6.6.0",
     "rn-fetch-blob": "^0.12.0",
-    "sendbird": "^3.0.119"
+    "sendbird": "^3.0.119",
+    "sinon": "^9.0.2"
```
Comment (kartikeya01) [inline package.json:39]: Why is this dependency necessary? Sometimes errors say you need to download dependencies but that is rarely the case.

File: package.json [hunk 9]
```diff
@@ -35,13 +35,15 @@
...
   },
   "devDependencies": {
     "@babel/core": "^7.8.4",
     "@babel/runtime": "^7.8.4",
     "@react-native-community/eslint-config": "^0.0.7",
     "babel-jest": "^25.1.0",
+    "check-prop-types": "^1.1.2",
```
Comment (kartikeya01) [inline package.json:46]: Why is this dependency necessary? Sometimes errors say you need to download dependencies but that is rarely the case.

File: src/screens/Feed.js (hunk 4 above)
Comment (kartikeya01) [inline src/screens/Feed.js:33]: We should add a console.log() statement here to make sure the TouchableWithoutFeedback works as intended

File: src/screens/app/Components/UserCaption/UserCaption.js [hunk 10]
```diff
@@ -1,23 +1,32 @@
 import React from 'react';
 import {Subheading} from 'react-native-paper';
 import {Text, StyleSheet} from 'react-native';
+import PropTypes from 'prop-types';
```
Comment (kartikeya01) [inline src/screens/app/Components/UserCaption/UserCaption.js:4]: We shouldn't be needing the dependency I think. It's usually a syntax error that makes you download the dependency

File: __tests__/UserCaption-test.js (hunk 2 above)
Comment (denny999222) [inline __tests__/UserCaption-test.js:26]: suggest use es6 syntax where you can do const {username, caption } = instance.state to reduce redundnacy

File: src/screens/app/Components/SettingsButtonVertical/SettingsButtonVertical.js (hunk 5 above)
Comment (denny999222) [inline src/screens/app/Components/SettingsButtonVertical/SettingsButtonVertical.js:15]: we can just name component <settingsbutton/> and pass a prop to give it an option to be vertical or horizontal

File: __tests__/UserCaption-test.js (hunk 1 above)
Comment (EramM805) [reply __tests__/UserCaption-test.js:5]: What do you mean by "this"? assert is a native js function and check-prop-types is a dependency I have added. You can find the link here : https://www.npmjs.com/package/check-prop-types

File: __tests__/UserCaption-test.js (hunk 2 above)
Comment (EramM805) [reply __tests__/UserCaption-test.js:26]: yea it should be a prop, not a state. Good catch

File: package.json (hunk 9 above)
Comment (EramM805) [reply package.json:46]: We do need check-prop-types, it makes unit testing for prop-types easier.

File: package.json (hunk 8 above)
Comment (EramM805) [reply package.json:39]: I was trying to figure out how to do unit testing with prop-types, and sinon was a solution. We are no longer using sinon, so I will delete it. Great catch.

File: src/screens/app/Components/UserCaption/UserCaption.js (hunk 10 above)
Comment (EramM805) [reply src/screens/app/Components/UserCaption/UserCaption.js:4]: I'll look into this. Pretty sure we need prop-types to use PropTypes!

File: src/screens/app/Components/SettingsButtonVertical/SettingsButtonVertical.js (hunk 5 above)
Comment (EramM805) [reply src/screens/app/Components/SettingsButtonVertical/SettingsButtonVertical.js:15]: great idea. I'll do this!

Comment (EramM805) [reply src/screens/app/Components/SettingsButtonVertical/SettingsButtonVertical.js:15]: This is a mistake. I am going to create just a custom settings button where we just add a prop, it's an idea from @denny999222

File: src/screens/app/Components/UserCaption/UserCaption.js (hunk 6 above)
Comment (EramM805) [reply src/screens/app/Components/UserCaption/UserCaption.js:14]: Will be converting it to props! Thanks again for noticing

File: src/screens/app/Components/UserCaption/UserCaption.js (hunk 7 above)
Comment (EramM805) [reply src/screens/app/Components/UserCaption/UserCaption.js:26]: yea I will be adding empty string as a test, it should work

File: __tests__/UserCaption-test.js (hunk 3 above)
Comment (EramM805) [reply __tests__/UserCaption-test.js:185]: Yes, great suggestion

File: src/screens/Feed.js (hunk 4 above)
Comment (EramM805) [reply src/screens/Feed.js:33]: Remove <TouchableWithoutFeedback onPress={() => { }}>

File: src/screens/app/Components/SettingsButtonVertical/SettingsButtonVertical.js (hunk 5 above)
Comment (EramM805) [reply src/screens/app/Components/SettingsButtonVertical/SettingsButtonVertical.js:15]: Created SettingsButton component with unit tests

File: src/screens/Feed.js (hunk 4 above)
Comment (EramM805) [reply src/screens/Feed.js:33]: Removed <TouchableWithoutFeedback onPress={() => {}}>

File: __tests__/UserCaption-test.js (hunk 2 above)
Comment (EramM805) [reply __tests__/UserCaption-test.js:26]: Created Function, so this no longer applies

File: src/screens/app/Components/SettingsButtonVertical/SettingsButtonVertical.js (hunk 5 above)
Comment (EramM805) [reply src/screens/app/Components/SettingsButtonVertical/SettingsButtonVertical.js:15]: Completed with unit tests!
